src/stub/core_impl/thread_extra_info.rs

```diff
@@ -0,0 +1,37 @@
+use super::prelude::*;
+use crate::protocol::commands::ext::ThreadExtraInfo;
+use crate::protocol::SpecificIdKind;
+use crate::target::ext::base::BaseOps;
+
+impl<T: Target, C: Connection> GdbStubImpl<T, C> {
+    pub(crate) fn handle_thread_extra_info<'a>(
+        &mut self,
+        res: &mut ResponseWriter<'_, C>,
+        target: &mut T,
+        command: ThreadExtraInfo<'a>,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let ops = match target.base_ops() {
+            BaseOps::SingleThread(_) => return Ok(HandlerStatus::Handled),
+            BaseOps::MultiThread(ops) => ops,
+        };
+
+        let handler_status = match command {
+            ThreadExtraInfo::qThreadExtraInfo(info) => {
+                if let Some(ops) = ops.support_thread_extra_info() {
+                    if let SpecificIdKind::WithId(tid) = info.id.tid {
+                        let size = ops.extra_info(tid, info.buf).handle_error()?;
+                        let data = &mut info.buf[..size];
```

Comment:
take a look at how `vFilePread` handles this. `gdbstub` takes care to avoid introducing any panicking paths, which includes those that might occur from array indexing.

Reply:
Updated